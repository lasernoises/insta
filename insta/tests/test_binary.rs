#[test]
fn test_binary_snapshot() {
    insta::assert_binary_snapshot!("txt", b"test".to_vec());
}

#[test]
#[should_panic(expected = "this file extension is not allowed")]
fn test_new_extension() {
    insta::assert_binary_snapshot!("new", b"test".to_vec());
}

#[test]
#[should_panic(expected = "file extensions starting with 'new.' are not allowed")]
fn test_extension_starting_with_new() {
    insta::assert_binary_snapshot!("new.gz", b"test".to_vec());
}

#[test]
fn test_multipart_extension() {
    insta::assert_binary_snapshot!("tar.gz", b"test".to_vec());
}
