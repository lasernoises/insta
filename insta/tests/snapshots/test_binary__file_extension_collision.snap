---
source: insta/tests/test_binary.rs
expression: "|file| { file.write_all(&[0, 1, 0, 1]).unwrap(); }"
extension: snap
snapshot_type: binary
---
